src/otx/algo/classification/timm_model.py

```diff
@@ -0,0 +1,299 @@
+        self.pretrained = pretrained
+
+        super().__init__(
+            label_info=label_info,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+            train_type=train_type,
+        )
+
+    def _create_model(self) -> nn.Module:
+        # Get classification_layers for class-incr learning
+        sample_model_dict = self._build_model(num_classes=5).state_dict()
+        incremental_model_dict = self._build_model(num_classes=6).state_dict()
+        self.classification_layers = get_classification_layers(
+            sample_model_dict,
+            incremental_model_dict,
+            prefix="model.",
+        )
+
+        model = self._build_model(num_classes=self.num_classes)
+        model.init_weights()
+        return model
+
+    def _build_model(self, num_classes: int) -> nn.Module:
+        backbone = TimmBackbone(backbone=self.backbone, pretrained=self.pretrained)
+        neck = GlobalAveragePooling(dim=2)
+        loss = nn.CrossEntropyLoss(reduction="none")
+        if self.train_type == OTXTrainType.SEMI_SUPERVISED:
+            return SemiSLClassifier(
+                backbone=backbone,
+                neck=neck,
+                head=OTXSemiSLLinearClsHead(
+                    num_classes=num_classes,
+                    in_channels=backbone.num_features,
+                    loss=loss,
+                ),
+            )
+
+        return ImageClassifier(
+            backbone=backbone,
+            neck=neck,
+            head=LinearClsHead(
+                num_classes=num_classes,
+                in_channels=backbone.num_features,
+                topk=(1, 5) if num_classes >= 5 else (1,),
+                loss=loss,
+            ),
+        )
+
+    def load_from_otx_v1_ckpt(self, state_dict: dict, add_prefix: str = "model.") -> dict:
+        """Load the previous OTX ckpt according to OTX2.0."""
+        return OTXv1Helper.load_cls_effnet_v2_ckpt(state_dict, "multiclass", add_prefix)
+
+    def forward_explain(self, inputs: MulticlassClsBatchDataEntity) -> MulticlassClsBatchPredEntity:
+        """Model forward explain function."""
+        outputs = self.model(images=inputs.stacked_images, mode="explain")
+
+        return MulticlassClsBatchPredEntity(
+            batch_size=len(outputs["preds"]),
+            images=inputs.images,
+            imgs_info=inputs.imgs_info,
+            labels=outputs["preds"],
+            scores=outputs["scores"],
+            saliency_map=outputs["saliency_map"],
+            feature_vector=outputs["feature_vector"],
+        )
+
+    def forward_for_tracing(self, image: torch.Tensor) -> torch.Tensor | dict[str, torch.Tensor]:
+        """Model forward function used for the model tracing during model exportation."""
+        if self.explain_mode:
+            return self.model(images=image, mode="explain")
+
+        return self.model(images=image, mode="tensor")
+
```

Comment:
It seems that these two functions are common in classification model. Then how about moving them to the parent class?

Reply:
These two functions are dependent on the `ImageClassifier`, so they don't apply to all classifications. I'll see if we can refactor `ImageClassifier` in the future to clean it up if possible.